angular/projects/admin-nrpti/src/app/mines/mines-detail/mines-detail.component.ts

```diff
@@ -15,55 +13,115 @@ import { FactoryService } from '../../services/factory.service';
 export class MinesDetailComponent implements OnInit, OnDestroy {
   private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
 
-  public record = null;
+  public mine = null;
+  public canPublish = false;
+  public isPublished = false;
 
   constructor(public route: ActivatedRoute,
               public router: Router,
               private factoryService: FactoryService,
-              public changeDetectionRef: ChangeDetectorRef,
-              private dialogService: DialogService) {
+              public changeDetectionRef: ChangeDetectorRef) {
   }
 
   ngOnInit() {
     this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
-      if (!res || !res.record) {
+      if (!res || !res.mine) {
         alert("Uh-oh, couldn't load Mine");
         this.router.navigate(['mines']);
         return;
       }
 
-      this.record = res.record[0] && res.record[0].data && new Mine(res.record[0].data);
+      this.mine = res.mine[0] && res.mine[0].data && new Mine(res.mine[0].data);
+
+      this.isPublished = this.isRecordPublished();
+      this.canPublish = this.checkCanPublish();
 
       this.changeDetectionRef.detectChanges();
     });
   }
 
-  delete() {
-    this.dialogService.addDialog(ConfirmComponent,
-    {
-      title: 'Confirm Deletion',
-      message: 'Do you really want to delete this Mine Item?',
-      okOnly: false
-    }, {
-      backdropColor: 'rgba(0, 0, 0, 0.5)'
-    })
-    .pipe(takeUntil(this.ngUnsubscribe))
-    .subscribe(
-      isConfirmed => {
-        if (isConfirmed) {
-          try {
-            this.factoryService.deleteMineItem(this.record._id, 'mine');
-            this.router.navigate(['mines']);
-          } catch (e) {
-            alert('Could not delete Mine Item');
-          }
+  isRecordPublished(): boolean {
+    return this.mine && this.mine.read && this.mine.read.includes('public');
+  }
+
+  checkCanPublish(): boolean {
+    return (this.mine.name
+      && this.mine.status
+      && this.mine.permitNumbers
+      && this.mine.permitNumbers.length > 0
+      && this.mine.tailingsImpoundments
+      && this.mine.commodities
+      && this.mine.commodities.length > 0
+      && this.mine.location
+      && this.mine.location.coordinates
+      && this.mine.location.coordinates.length > 0);
+  }
+
+  publish(): void {
+    this.factoryService
+      .publishRecord(this.mine)
+      .pipe(
+        takeUntil(this.ngUnsubscribe),
+        catchError(error => {
+          console.log('Publish error:', error);
+          alert('Failed to publish record.');
+          return of(null);
+        })
+      )
+      .subscribe(response => {
+        if (!response) {
+          return;
         }
-      }
-    );
+
+        if (response.code === 409) {
+          // object was already published
```

Comment:
Does the server respond with more information about this? It should... if it does we should probably let hte user know why it failed.

Reply:
I'll look into this